### PR TITLE
Added GUI model.

### LINKED ISSUE #85: GUI Model
A model of the GUI shall be created. This does **not** have to be a sketch. A textual description of the important parts is sufficient.

### PR DESCRIPTION
**Pull request by RomanLangrehr**
Fixes #85

[PDF](http://romanlangrehr.github.io/Beagle/branches/GUI/Requirements%20Specification.pdf)

### CODE REVIEW

**Comment by flyfl** [comment]
Lgtm! I like that GUI model quite much. Was all of this decided in the meeting?

**Comment by RomanLangrehr** [comment]
> Was all of this decided in the meeting?
yes

**Comment by jGleitz** [comment]
There are some points in there that are different from what we discussed in the meeting. Unfortunately, reviewing on mobile is almost impossible. 
Please don't merge, I'll do a proper review tomorrow, as soon as possible.

**Comment by viridispirus** [comment]
very well, 
lgtm

**Comment by m1cm1c** [comment]
@RomanLangrehr Conflicts.

File: doc/Requirements Specification/Parts/Non-Functional Requirements.lyx [hunk 1]
```diff
@@ -430,5 +430,274 @@ gls{SoMoX}
...
+\begin_layout Plain Layout
+
+
+\backslash
+lstinline{.repository_diagram}
+\end_layout
+
+\end_inset
+
+ file in eclipse.
```
Comment (jGleitz) [inline doc/Requirements Specification/Parts/Non-Functional Requirements.lyx]: Yesterday, we discussed that these entries are in the project’s context menu and maybe in the global “Project” menu entry.

File: doc/Requirements Specification/Parts/Non-Functional Requirements.lyx [hunk 2]
```diff
@@ -430,5 +430,274 @@ gls{SoMoX}
...
+\backslash
+gls{component}
+\end_layout
+
+\end_inset
+
+ in the repository diagram, where an analysis with Beagle is possible.
+ Otherwise the context menu entry will be shown greyed and a description
+ why the analysis is not possible appears, when the users tries to start
+ the analysis.
```
Comment (jGleitz) [inline doc/Requirements Specification/Parts/Non-Functional Requirements.lyx]: These constraints also apply if launching Beagle for a whole project or a single internal action.

File: doc/Requirements Specification/Parts/Non-Functional Requirements.lyx [hunk 3]
```diff
@@ -430,5 +430,274 @@ gls{SoMoX}
...
+\end_layout
+
+\end_inset
+
+ has finished.
+\end_layout
+
+\begin_layout Standard
+When the users launches the analysis by a context menu (option 1, 2 or 3),
+ he gets an window where he can change some settings for Beagle:
```
Comment (jGleitz) [inline doc/Requirements Specification/Parts/Non-Functional Requirements.lyx]: He also has this options when using option 4 (when configuring SoMoX)

File: doc/Requirements Specification/Parts/Non-Functional Requirements.lyx (hunk 1 above)
Comment (RomanLangrehr) [inline doc/Requirements Specification/Parts/Non-Functional Requirements.lyx]: Yes, but I no plugin directly modifies the global project menu entry. In my opinion it's strange, when our, rather small, plugin has it's entries there while important plugins as Palladio don't.

File: doc/Requirements Specification/Parts/Non-Functional Requirements.lyx (hunk 2 above)
Comment (RomanLangrehr) [inline doc/Requirements Specification/Parts/Non-Functional Requirements.lyx]: Ok, I'll change this.

File: doc/Requirements Specification/Parts/Non-Functional Requirements.lyx (hunk 3 above)
Comment (RomanLangrehr) [inline doc/Requirements Specification/Parts/Non-Functional Requirements.lyx]: But it's another window there, isn't it?
I thought we just add some GUI elements to a SoMoX window...

File: doc/Requirements Specification/Parts/Non-Functional Requirements.lyx (hunk 1 above)
Comment (jGleitz) [inline doc/Requirements Specification/Parts/Non-Functional Requirements.lyx]: I see your point.

**Pull request merged**